src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -78,22 +78,41 @@ class PromotionProvider(private val project: Project) {
                     val notifications = getSonarLintProjectNotifications()
                     val wasAutoAnalyzed = PropertiesComponent.getInstance().getLong(FIRST_AUTO_ANALYSIS_DATE, 0L) != 0L
 
-                    if (!Settings.getSettingsFor(project).isBound && !Settings.getGlobalSettings().isPromotionDisabled && wasAutoAnalyzed) {
-                        processFullProjectPromotion(notifications)
-                    }
-
-                    if (nonReportAnalysisTriggers.contains(triggerType) && !wasAutoAnalyzed) {
-                        PropertiesComponent.getInstance().setValue(FIRST_AUTO_ANALYSIS_DATE, Instant.now().toEpochMilli().toString())
+                    if (nonReportAnalysisTriggers.contains(triggerType)) {
+                        processAutoAnalysisTriggers(wasAutoAnalyzed, notifications)
                     }
 
                     if (reportAnalysisTriggers.contains(triggerType)) {
-                        PropertiesComponent.getInstance().setValue(WAS_REPORT_EVER_USED, true)
+                        processReportAnalysisTriggers(files, notifications)
                     }
```

Comment:
I would expect those 2 `if`s to be moved inside the `isPromotionEnabled()`, I think it would make more sense. Do you see any issue with that?